### PR TITLE
Fix mypy error in pandas\tests\indexes\interval\test_interval_tree.py

### PR DESCRIPTION
**Pull request by pqzx**
- [x] relates to #28926
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/indexes/interval/test_interval_tree.py [hunk 1]
```diff
@@ -154,14 +154,14 @@ def test_get_indexer_closed(self, closed, leaf_size):
             (np.array([0, 2, np.nan]), np.array([1, 3, np.nan]), False),
         ],
     )
-    @pytest.mark.parametrize("order", map(list, permutations(range(3))))
+    @pytest.mark.parametrize("order", map(lambda x: list(x), permutations(range(3))))
```
Comment (simonjayhawkins) [inline pandas/tests/indexes/interval/test_interval_tree.py]: as this is a known issue with mypy, see https://github.com/python/mypy/issues/6697, i'd prefer not to change the code here.

add a `# type: ignore` and add the issue number as a comment i.e.

```python
    # https://github.com/python/mypy/issues/6697
    # error: Argument 1 to "map" has incompatible type "Type[List[Any]]";
    #  expected "Callable[[Tuple[int, ...]], List[_T]]"
    @pytest.mark.parametrize("order", map(list, permutations(range(3))))  # type: ignore
```

File: pandas/tests/indexes/interval/test_interval_tree.py [hunk 2]
```diff
@@ -154,14 +154,14 @@ def test_get_indexer_closed(self, closed, leaf_size):
...
     def test_is_overlapping(self, closed, order, left, right, expected):
         # GH 23309
         tree = IntervalTree(left[order], right[order], closed=closed)
         result = tree.is_overlapping
         assert result is expected
 
-    @pytest.mark.parametrize("order", map(list, permutations(range(3))))
+    @pytest.mark.parametrize("order", map(lambda x: list(x), permutations(range(3))))
```
Comment (simonjayhawkins) [inline pandas/tests/indexes/interval/test_interval_tree.py]: same

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]
@pqzx Thanks for the PR.

**Review by simonjayhawkins — APPROVED** [review]
@pqzx lgtm @WillAyd

**Comment by WillAyd** [comment]
Hmm just out of curiosity does this work if you did `(list(x) for x in permutations(range(3)))`. Assuming equivalent I think would be easier to track than the type ignore

**Comment by pqzx** [comment]
> 
> 
> Hmm just out of curiosity does this work if you did `(list(x) for x in permutations(range(3)))`. Assuming equivalent I think would be easier to track than the type ignore

Yes, this works. Updated to use this instead.

**Comment by simonjayhawkins** [comment]
@pqzx can you fix lint error and merge master

**Comment by pqzx** [comment]
@simonjayhawkins sorry for the noob question, what's the lint error in this case? Is it the line length?

**Comment by WillAyd** [comment]
Looks like the checks are fine. I think if you merge master and try again should work (have had intermittent failures recently)

**Comment by simonjayhawkins** [comment]
@WillAyd OK to merge?

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @pqzx !

**Pull request merged**